grid_map_ros/test/GridMapRosTest.cpp

```diff
@@ -151,35 +169,133 @@ TEST(OccupancyGridConversion, roundTrip)
   occupancyGrid.info.origin.orientation.w = 1.0;
   occupancyGrid.data.resize(occupancyGrid.info.width * occupancyGrid.info.height);
 
-  for (auto& cell : occupancyGrid.data) {
-    cell = rand() % 102 - 1; // [-1, 100]
+  unsigned int seed = time(0);
+  for (auto & cell : occupancyGrid.data) {
+    cell = rand_r(&seed) % 102 - 1;  // [-1, 100]
   }
 
   // Convert to grid map.
   GridMap gridMap;
   GridMapRosConverter::fromOccupancyGrid(occupancyGrid, "layer", gridMap);
 
   // Convert back to occupancy grid.
-  nav_msgs::OccupancyGrid occupancyGridResult;
+  nav_msgs::msg::OccupancyGrid occupancyGridResult;
   GridMapRosConverter::toOccupancyGrid(gridMap, "layer", -1.0, 100.0, occupancyGridResult);
 
   // Check map info.
   EXPECT_EQ(occupancyGrid.header.stamp, occupancyGridResult.header.stamp);
   EXPECT_EQ(occupancyGrid.header.frame_id, occupancyGridResult.header.frame_id);
   EXPECT_EQ(occupancyGrid.info.width, occupancyGridResult.info.width);
   EXPECT_EQ(occupancyGrid.info.height, occupancyGridResult.info.height);
-  EXPECT_DOUBLE_EQ(occupancyGrid.info.origin.position.x, occupancyGridResult.info.origin.position.x);
-  EXPECT_DOUBLE_EQ(occupancyGrid.info.origin.position.x, occupancyGridResult.info.origin.position.x);
-  EXPECT_DOUBLE_EQ(occupancyGrid.info.origin.orientation.x, occupancyGridResult.info.origin.orientation.x);
-  EXPECT_DOUBLE_EQ(occupancyGrid.info.origin.orientation.y, occupancyGridResult.info.origin.orientation.y);
-  EXPECT_DOUBLE_EQ(occupancyGrid.info.origin.orientation.z, occupancyGridResult.info.origin.orientation.z);
-  EXPECT_DOUBLE_EQ(occupancyGrid.info.origin.orientation.w, occupancyGridResult.info.origin.orientation.w);
+  EXPECT_DOUBLE_EQ(
+    occupancyGrid.info.origin.position.x,
+    occupancyGridResult.info.origin.position.x);
+  EXPECT_DOUBLE_EQ(
+    occupancyGrid.info.origin.position.x,
+    occupancyGridResult.info.origin.position.x);
+  EXPECT_DOUBLE_EQ(
+    occupancyGrid.info.origin.orientation.x,
+    occupancyGridResult.info.origin.orientation.x);
+  EXPECT_DOUBLE_EQ(
+    occupancyGrid.info.origin.orientation.y,
+    occupancyGridResult.info.origin.orientation.y);
+  EXPECT_DOUBLE_EQ(
+    occupancyGrid.info.origin.orientation.z,
+    occupancyGridResult.info.origin.orientation.z);
+  EXPECT_DOUBLE_EQ(
+    occupancyGrid.info.origin.orientation.w,
+    occupancyGridResult.info.origin.orientation.w);
 
   // Check map data.
   for (std::vector<int8_t>::iterator iterator = occupancyGrid.data.begin();
-      iterator != occupancyGrid.data.end(); ++iterator) {
+    iterator != occupancyGrid.data.end(); ++iterator)
+  {
     size_t i = std::distance(occupancyGrid.data.begin(), iterator);
-    EXPECT_EQ((int)*iterator, (int)occupancyGridResult.data[i]);
+    EXPECT_EQ(static_cast<int>(*iterator), static_cast<int>(occupancyGridResult.data[i]));
+  }
+}
+
+TEST(CostmapConversion, withMove)
```

Comment:
Are these new unit tests?

Reply:
Yeah, he added both nav_msgs/occupancygrid and nav2_msgs/costmap conversions. We use nav2_msgs/costmap in a couple places to publish raw costmaps for other processes to use (primarily for recovery actions that need some view of the world but aren't in the same process as the local costmap / controller).

Reply:
That is very cool, thanks that you added them.